web/src/app/services/auth.service.ts

```diff
@@ -1,21 +1,117 @@
 import { Injectable } from '@angular/core';
 import { AngularFireAuth } from 'angularfire2/auth';
+import { AngularFirestore } from 'angularfire2/firestore';
 import * as firebase from 'firebase/app';
-import { Observable, from } from 'rxjs';
+import { Observable, from, of } from 'rxjs';
+import { map, catchError, switchMap } from 'rxjs/operators';
+import { ErrorService } from './error.service';
+import { User } from '../models/user.model';
+
 @Injectable()
 export class AuthService {
-  constructor(private firebaseAuth: AngularFireAuth) {
+
+  user: Observable<User>;
+
+  constructor(
+    private firebaseAuth: AngularFireAuth,
+    private errorService: ErrorService,
+    private firestore: AngularFirestore
+  ) {
+    this.user = this.firebaseAuth.authState.pipe(
+      switchMap(user => {
+        if (user) {
+          return this.firestore.doc<User>(`user/${user.uid}`).snapshotChanges();
+        } else {
+          return of(null);
+        }
+      })
+    );
   }
 
-  signInWithGithub(): Observable<any> {
+  signInWithGithub(): Observable<User> {
     return from(this.firebaseAuth.auth.signInWithPopup(
-      new firebase.auth.GithubAuthProvider()
-    ));
+      new firebase.auth.GithubAuthProvider()))
+      .pipe(
+        map((user) => this.formatUserResponse(user, 'github')),
+        catchError((error) => this.errorService.logError(error))
+      );
   }
 
-  signInWithTwitter(): Observable<any> {
-    return from(this.firebaseAuth.auth.signInWithPopup(
-      new firebase.auth.TwitterAuthProvider()
-    ));
+  signInWithTwitter() {
```

Comment:
Missing return type

Reply:
As I am currently working on formatting the response of twitter. I have not yet added the return type for this function

Reply:
Ok. Will the return type be done in this PR?

Reply:
No Will add it in the next PR

Reply:
Ok